File: app/lib/models/track.dart
class Track {
  final String id;
  final String title;
  final String artist;
  final String filename;
  final String uploadedAt;

  const Track({
    required this.id,
    required this.title,
    required this.artist,
    required this.filename,
    required this.uploadedAt,
  });

  factory Track.fromJson(Map<String, dynamic> json) {
    return Track(
      id: json['id'] as String,
      title: json['title'] as String,
      artist: json['artist'] as String? ?? '',
      filename: json['filename'] as String,
      uploadedAt: json['uploaded_at'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'artist': artist,
      'filename': filename,
      'uploaded_at': uploadedAt,
    };
  }
}

File: server/src/models/track.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Track {
    pub id: Uuid,
    pub title: String,
    pub artist: String,
    pub filename: String,
    pub filepath: String,
    pub uploaded_at: DateTime<Utc>,
}
